src/web/modules/gateway_accounts/csv_data.ts

```diff
@@ -3,28 +3,20 @@ import { AdminUsers, Connector } from '../../../lib/pay-request'
 import { Service } from '../../../lib/pay-request/types/adminUsers'
 import { aggregateServicesByGatewayAccountId, extractFiltersFromQuery, toAccountSearchParams, Filters } from '../../../lib/gatewayAccounts'
 import { GatewayAccount } from '../../../lib/pay-request/types/connector'
-import { parse } from 'qs'
-import { Request } from 'express'
 
 async function getServiceGatewayAccountIndex(): Promise<{ [key: string]: Service }> {
   const services = await AdminUsers.services()
   return aggregateServicesByGatewayAccountId(services)
 }
 
-async function getAccounts(filters: Filters): Promise<GatewayAccount[]> {
+function getAccounts(filters: Filters): Promise<GatewayAccount[]> {
   const searchParams = toAccountSearchParams(filters)
-  return await Connector.accounts(searchParams).then((response: any) => response.accounts)
+  return Connector.accounts(searchParams).then((response: any) => response.accounts)
 }
 
-function getFiltersFromQuery(req: Request): Filters {
-  const query = parse(req.query)
-  return extractFiltersFromQuery(query)
-}
-
-export async function createCsvWithAdminEmailsData(req: Request): Promise<any> {
-  const filters = getFiltersFromQuery(req)
-  const [accountsResponse, serviceGatewayAccountIndex] = 
-    await Promise.all([await getAccounts(filters), await getServiceGatewayAccountIndex()])
+export async function createCsvWithAdminEmailsData(filters: Filters): Promise<any> {
```

Comment:
Nice 👍  definitely favour only passing what methods need, much easier to read and less scope for unintended side effects